lib/ash/resource/preparation/preparation.ex

```diff
@@ -5,6 +5,8 @@ defmodule Ash.Resource.Preparation do
   `c:init/1` is defined automatically by `use Ash.Resource.Preparation`, but can be implemented if you want to validate/transform any
   options passed to the module.
 
+  `c:supports/1` is also required and returns a list containing any or all of the following: `[Ash.Changeset, Ash.Query, Ash.ActionInput]` based on what your `c:prepare/3` implementation supports
```

Comment:
required is nebulous because its not required to implement yourself. A default is provided by the `use` macro.